### PR TITLE
[Feature] Add elementwise_affine argument to LigerRMSNorm

### PR DESCRIPTION
**Pull request by niyunsheng**
## Summary

This PR adds the `elementwise_affine` argument to `LigerRMSNorm` to align its API with `torch.nn.RMSNorm`. This allows users to instantiate the normalization layer without learnable parameters.

## Details
- Added `elementwise_affine` (defaulting to `True`) to the `__init__` method of `LigerRMSNorm`.
- When `elementwise_affine` is set to `False`, `self.weight` is registered as `None`, preventing the allocation of unnecessary parameters.
- Updated `extra_repr` to include the `elementwise_affine` status in the printed model structure.
- The parameter is added at the end of the argument list to maintain backward compatibility for subclasses (e.g., `LigerRMSNormForGemma`) that rely on positional arguments during `super().__init__` calls.

## Testing Done

- Hardware Type: NVIDIA A100-SXM4-80GB
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

### CODE REVIEW

File: test/transformers/test_rms_norm.py [hunk 1]
```diff
@@ -128,19 +149,33 @@ def test_correctness(bs, sl, hd, dtype, atol, rtol, reference, offset, casting_m
     do = torch.randn(bs, sl, hd, device=device, dtype=dtype)
 
     # reference (llama or gemma)
-    ref_rms = reference(hidden_size=hd).to(device).to(dtype)
+    if not elementwise_affine and reference == GemmaRMSNorm:
+        return pytest.skip("GemmaRMSNorm does not support elementwise_affine=False")
```
Comment (Tcc0403) [inline test/transformers/test_rms_norm.py]: Why doesn't GemmaRMSNorm support it?

Comment (niyunsheng) [reply test/transformers/test_rms_norm.py]: Because GemmaRMSNorm with elementwise_affine=False will be LigerRMSNorm

Comment (niyunsheng) [reply test/transformers/test_rms_norm.py]: > Why doesn't GemmaRMSNorm support it?

@Tcc0403  I believe adding support within this test file is a better approach and will reduce confusion. I’ll push a new commit shortly.

Comment (niyunsheng) [reply test/transformers/test_rms_norm.py]: > > Why doesn't GemmaRMSNorm support it?
> 
> @Tcc0403 I believe adding support within this test file is a better approach and will reduce confusion. I’ll push a new commit shortly.

@Tcc0403 Done.
